CorsixTH/Src/main.cpp

```diff
@@ -22,17 +22,30 @@ SOFTWARE.
 
 #include "config.h"
 
+#include <array>
 #include <cstdio>
 #include <cstring>
 #include <string>
 
+#ifdef _WIN32
```

Comment:
I don't love the idea of depending on (and assuming POSIX) everywhere outside of Windows.

We don't have C++ 17's `std::filesystem::exists` but we can detect what we really care about, whether we can read a file pretty easily with ifstream.

```
namespace {
bool file_exists(std::string f) {
  std::ifstream file(f);
  return file.is_open();
}
}
```

Reply:
My comment was meant as a quickly written reference rather than an exact template.

Take a reference instead of a value (std::string&), also remove the .cstr()'s where you're calling this function.